test/e2e/json-rpc/switchEthereumChain.spec.js

```diff
@@ -278,7 +326,36 @@ describe('Switch Ethereum Chain for two dapps', function () {
           text: 'Connect',
           tag: 'button',
         });
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+        await driver.clickElement(
+          '[data-testid ="account-options-menu-button"]',
+        );
+        await driver.clickElement({ text: 'All Permissions', tag: 'div' });
+        await driver.clickElementAndWaitToDisappear({
+          text: 'Got it',
+          tag: 'button',
+        });
+        await driver.clickElement({
+          text: '127.0.0.1:8081',
+          tag: 'p',
+        });
+        const editButtons = await driver.findElements('[data-testid="edit"]');
 
+        // Ensure there are edit buttons
+        assert.ok(editButtons.length > 0, 'Edit buttons are available');
+
+        // Click the first (0th) edit button
+        await editButtons[1].click();
+
+        // Disconnect Mainnet
+        await driver.clickElement({
+          text: 'Localhost 8545',
+          tag: 'p',
+        });
+
+        await driver.clickElement('[data-testid="connect-more-chains-button"]');
```

Comment:
same here, would be nice to just not connect Dapp Two to 0x539 during the connection flow itself rather than disconnecting it afterwards

Reply:
Yeah, updated the connection logic to remove 0x539 in initial connection screen https://github.com/MetaMask/metamask-extension/pull/27980/commits/bd26be26bb958d6ffe745d6b76ec79d57885a06f